src/main.rs

```diff
@@ -89,43 +89,44 @@ async fn main() -> Result<()> {
             &args.wg_interface,
             peers,
         )?;
-        networkd_config.write_config(&args.networkd_dir).await?;
+        networkd_config
+            .write_config(&args.networkd_dir, args.keepalive)
+            .await?;
         info!("Our new config is:\n{:#?}", networkd_config);
     }
 
     info!("Restarting systemd-networkd");
     NetworkdConfiguration::restart().await?;
     consul_client.drop_lock().await?;
 
-    // Enter main loop which periodically checks for updates to the list of WireGuard peers.
-    // To make sure we give new peers a chance to send handshakes, we need to debounce
-    // timeout removals.
-    let mut last_timeout_removal = Instant::now();
-    let timeout_debounce = Duration::from_secs(60);
+    // Stores amount of received bytes together with an instant for every peer.
+    let mut received_bytes: HashMap<wireguard_keys::Pubkey, (usize, Instant)> = HashMap::new();
 
+    // Enter main loop which periodically checks for updates to the list of WireGuard peers.
     loop {
-        if args.peer_timeout > Duration::ZERO && last_timeout_removal.elapsed() > timeout_debounce {
```

Comment:
It should still be possible to disable timeout entirely by setting the duration to zero. Please put this back in.

Reply:
Done.